app/javascript/show_search_fields.js

```diff
@@ -0,0 +1,44 @@
+document.addEventListener('turbo:load', () => {
+  const searchFields = document.querySelectorAll('[data-component="search-field"]')
+  searchFields.forEach( searchField => setupContainer(searchField))
+})
+
+const setupContainer = (searchField) => {
+  searchField.addEventListener(['keyup'], () => {
+    const searchText = searchField.querySelector('[data-search="search-input"]').value
+    showSearchResults(searchText, searchField)
+  })
+}
+
+const showSearchResults = (searchText, searchField) => {
+  const searchContainer = searchField.querySelector('[data-target="search-container"]')
+  if (searchText.length > 1) {
+    searchContainer.classList.remove('hidden')
+    const patientsList = document.getElementById('patients-list');
+    fetchPatients(searchText).then(patients => {
+      patientsList.innerHTML = ''
+      setupPatientsList(patients)
+    })
+  } else {
+    searchContainer.classList.add('hidden')
+  }
+}
+
+const fetchPatients = async (searchText) => {
+  const patients = await fetch(`/patients.json?text=${searchText}`)
```

Comment:
This represents a response, not `patients`

![Screenshot 2023-03-04 at 17 06 16](https://user-images.githubusercontent.com/26292224/222932637-4c856e06-9a60-44e7-9385-03cc3c761a14.png)


Please refer tot the [fetch API docs](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch). There you can find the right wait to do things here